### PR TITLE
Feature/verify access

### PR DESCRIPTION
**Pull request by jeffjohnson9046**
- Adds `isInSimulationEnvironment()` and `isInDevelopmentEnvironment()` methods on `ExternalSessionConfiguration` to identify if the environment is configured for "Development" or "SIMULATION"
- Adds `isProctorless()` to the `Session` to identify if the session is managed by a Proctor

### CODE REVIEW

File: client/src/main/java/tds/session/ExternalSessionConfiguration.java [hunk 1]
```diff
@@ -58,4 +61,22 @@ public int getShiftWindowStart() {
...
         return shiftWindowEnd;
     }
+
+    /**
+     * Determine if this {@link ExternalSessionConfiguration} is configured for the Simulation environment.
+     *
+     * @return True if the environment is set to "simulation" (case-insensitive); otherwise false.
+     */
+    public boolean isInSimulationEnvironment() {
+        return this.getEnvironment().equalsIgnoreCase(SIMULATION_ENVIRONMENT);
```
Comment (millerts) [inline client/src/main/java/tds/session/ExternalSessionConfiguration.java:79]: I believe this will throw if `environment` is null so add a null check?  Here and below.

**Review by millerts — CHANGES_REQUESTED** [review]

File: client/src/main/java/tds/session/ExternalSessionConfiguration.java (hunk 1 above)
Comment (jtreuting) [reply client/src/main/java/tds/session/ExternalSessionConfiguration.java:79]: Base don that, I think you should add a unit test for a NULL environment as well.

Comment (jeffjohnson9046) [reply client/src/main/java/tds/session/ExternalSessionConfiguration.java:79]: The `environment` cannot be `null` in either the `session._externs` or the `configs.client_externs` table, so the only way I can see `environment` getting set to `null` is when someone calls `ExternalSessionConfiguration` constructor.  Having said that: a `null` check here is a good safeguard, so I'll add it and a unit test to make sure it works.

**Comment by millerts** [comment]
Another option if it can never be null is to throw an `IllegalArgumentException` if the constructor is called with null.
